src/CommunityToolkit.Maui.SourceGenerators/Helpers/SourceStringExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Text;
+
+namespace CommunityToolkit.Maui.SourceGenerators.Helpers
+{
+	static class SourceStringExtensions
+	{
+		public static void FormatText(ref string classSource, CSharpParseOptions? options)
+		{
+			var mysource = CSharpSyntaxTree.ParseText(SourceText.From(classSource, Encoding.UTF8), options);
+			var formattedRoot = (CSharpSyntaxNode)mysource.GetRoot().NormalizeWhitespace();
```

Comment:
If you're already producing the generated source as a string and it's well-formatted (or you can make it well-formatted if it's not), then you don't really need to format the code via `NormalizeWhitespace`, nor you need to parse the source.

`NormalizeWhitespace` can be useful only if you're producing the code via `SyntaxFactory` APIs or something similar.

Reply:
Since this will be a method to be used by future SGs I will keep this way